conda/notices/views.py

```diff
@@ -15,17 +17,27 @@ def print_notices(channel_notices: Sequence[ChannelNotice]):
     """
     current_channel = None
 
-    for channel_notice in channel_notices:
-        if current_channel != channel_notice.channel_name:
+    if context.json:
+        json_output = json.dumps(
+            [
+                ChannelNotice.notice_to_json_notice(channel_notice)
```

Comment:
This isn't a correct use of the instance method you defined above, you're passing in an instance, while all that would be needed is calling it on the instance you already have here:
```suggestion
                channel_notice.to_dict()
```